staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/appendixa_test.go

```diff
@@ -289,11 +289,11 @@ func TestAppendixA(t *testing.T) {
 			},
 		},
 		{
-			example: hex("d74401020304"),
-			decoded: "\x01\x02\x03\x04",
-			encoded: hex("4401020304"),
+			example: hex("d74401020304"), // 23(h'01020304')
```

Comment:
what does this comment mean? is the 23 the expected format tag? does that map to base16 or am I misreading something? should this have been 22?

Reply:
The CBOR spec includes a human-readable diagnostic notation (https://www.rfc-editor.org/rfc/rfc8949.html#section-8). This comment is the diagnostic notation of `0xd74401020304`.

All of the encoded examples in this test come directly from https://www.rfc-editor.org/rfc/rfc8949.html#section-appendix.a. This one would not be produced by the CBOR serializer in apimachinery but it is understood and handled properly. Yes, 23 is "expected base16 encoding" (https://www.rfc-editor.org/rfc/rfc8949.html#name-expected-later-encoding-for), so when we decode the example to a string the bytes get converted to base16.

Reply:
ah, ok ... so the re-encoded bit is *not* using our encoder which would decorate with ByteSliceLaterFormatBase64?

Reply:
It's using that encoder, but it's encoding a Go string and not a []byte at that point. The sequence is:

1. Original bytes are 0xd74401020304. We'd hypothetically produce this if we were configured to used `ByteSliceLaterFormatBase16` and we were marshalling `[]byte{0x01, 0x02, 0x03, 0x04}`.
2. When unmarshalling into `interface{}`, it applies the text encoding indicated by tag 23 to the bytes. The dynamic value of the interface becomes the Go string `"01020304"`.
3. The value being marshalled back to CBOR is the string `"01020304"`.

This is consistent with JSON -- if you marshal a `[]byte`, you'll get the base64 encoding of those bytes in a JSON string. Unmarshalling that JSON string into a Go interface{} would give you a Go string containing the base64-encoded bytes.